vsphere/resource_vsphere_snapshot.go

```diff
@@ -0,0 +1,186 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func resourceVSphereSnapshot() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereSnapshotCreate,
+		Read:   resourceVSphereSnapshotRead,
+		Delete: resourceVSphereSnapshotDelete,
+
+		Schema: map[string]*schema.Schema{
+			"vm_id": {
```

Comment:
Can you change this to reference the the UUID of the virtual machine versus the VM path name?

`vsphere_virtual_machine` supplies this via the `uuid` attribute, so it can be fetched in Terraform. From there, you can get the MoRef for the VM via the [`FindAllByUuid`](https://godoc.org/github.com/vmware/govmomi/object#SearchIndex.FindByUuid) method.